src/sagemaker/jumpstart/utils.py

```diff
@@ -878,3 +880,99 @@ def get_region_fallback(
         return constants.JUMPSTART_DEFAULT_REGION_NAME
 
     return list(combined_regions)[0]
+
+
+def get_preset_names(
+    region: str,
+    model_id: str,
+    model_version: str,
+    sagemaker_session: Optional[Session] = constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+    scope: enums.JumpStartScriptScope = enums.JumpStartScriptScope.INFERENCE,
+    model_type: enums.JumpStartModelType = enums.JumpStartModelType.OPEN_WEIGHTS,
+) -> List[str]:
+    """Returns a list of preset names for the given model ID and region."""
+    model_specs = verify_model_region_and_return_specs(
+        region=region,
+        model_id=model_id,
+        version=model_version,
+        sagemaker_session=sagemaker_session,
+        scope=scope,
+        model_type=model_type,
+    )
+
+    if scope == enums.JumpStartScriptScope.INFERENCE:
+        presets = model_specs.inference_presets
+
+    if scope == enums.JumpStartScriptScope.TRAINING:
+        presets = model_specs.training_presets
+
+    return list(presets.preset_configs.keys()) if presets else []
+
+
+def get_benchmark_stats(
+    region: str,
+    model_id: str,
+    model_version: str,
+    config_names: Optional[List[str]] = None,
+    sagemaker_session: Optional[Session] = constants.DEFAULT_JUMPSTART_SAGEMAKER_SESSION,
+    scope: enums.JumpStartScriptScope = enums.JumpStartScriptScope.INFERENCE,
+    model_type: enums.JumpStartModelType = enums.JumpStartModelType.OPEN_WEIGHTS,
+) -> Dict[str, List[JumpStartBenchmarkStat]]:
+    """Returns benchmark stats for the given model ID and region."""
+    model_specs = verify_model_region_and_return_specs(
+        region=region,
+        model_id=model_id,
+        version=model_version,
+        sagemaker_session=sagemaker_session,
+        scope=scope,
+        model_type=model_type,
+    )
+
+    if scope == enums.JumpStartScriptScope.INFERENCE:
+        presets = model_specs.inference_presets
+    elif scope == enums.JumpStartScriptScope.TRAINING:
+        presets = model_specs.training_presets
+
+    if not config_names:
+        config_names = presets.preset_configs.keys() if presets else []
+
+    benchmark_stats = {}
+    for config_name in config_names:
+        if config_name not in presets.preset_configs:
+            raise ValueError(f"Unknown preset config name: '{config_name}'")
+        benchmark_stats[config_name] = presets.preset_configs.get(config_name).benchmark_metrics
```

Comment:
i think you need to do `get(config_name, {})`, or else you'll get an error if the key isn't there